### DISCUSSION TITLE
Is it possible to define which exact pages to build?

### DISCUSSION THREAD

**Discussion by ar-arkhipov** [11ty/eleventy#1745] [Q&A]
I wonder if it is possible (and how) to tell Eleventy to build only particular page or set of pages? For example build only particular page (specified via permalink ?), or build all (or define number) pages from the posts section (`/posts/*`), etc.

I can dig into Eleventy source code, but if someone already has an answer it would be great. Thanks!

**Comment by Ryuno-Ki** [comment]
Closest I can think of is the [input parameter](https://www.11ty.dev/docs/config/#input-directory).

**Comment by pdehaan** [comment]
I think some people have cleverly used the [.eleventyignore](https://www.11ty.dev/docs/ignores/) file in the past to only build a subset of their site during local development (where you might only want to build 10 blog posts vs 2000+).

**Reply by pdehaan** [reply]
https://github.com/11ty/eleventy/issues/1123 might also have some tips. You could possibly use a global eleventyComputed file or data directory file which checks for an environment variable and optionally sets a bunch of templates to `permalink: false`.

**Reply by edwardhorsford** [reply]
Unfortunately setting `permalink: false` still results in the files being processed. As a result you don't actually save much time on a large site - the full site data is still being worked through.

**Reply by nhoizey** [reply]
What about setting `eleventyExcludeFromCollections` to `false` in addition? I think it might help.

**Comment by ar-arkhipov** [comment]
The idea is to be able to run parallel builds for the same website. For example, if I have website consisting of 100 000 pages I would like to be able to execute 20 parallel builds (for example in AWS lambda) where each build processes 5000 pages. But at the same time I want to keep the website structure, pagination, etc., where in the end I can upload all the final files from 20 workers in the one S3 bucket. That is what I am trying to achieve.

**Reply by coryasilva** [reply]
I am thinking this could work:
lambda to collect page data --> kinesis data stream --> lambda build page --> s3 --> cloudfront

What is the source of your data? Static files or API?
